core/src/main/java/io/questdb/griffin/engine/functions/catalogue/AttributeCatalogueFunctionFactory.java

```diff
@@ -0,0 +1,239 @@
+import io.questdb.cairo.sql.*;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.CursorFunction;
+import io.questdb.std.*;
+import io.questdb.std.str.NativeLPSZ;
+import io.questdb.std.str.Path;
+
+public class AttributeCatalogueFunctionFactory implements FunctionFactory {
+
+    private static final RecordMetadata METADATA;
+
+    @Override
+    public String getSignature() {
+        return "pg_catalog.pg_attribute()";
+    }
+
+    @Override
+    public Function newInstance(ObjList<Function> args, int position, CairoConfiguration configuration) {
+        return new CursorFunction(
+                position,
+                new AttributeCatalogueCursorFactory(
+                        configuration,
+                        METADATA
+                )
+        );
+    }
+
+    private static class AttributeCatalogueCursorFactory extends AbstractRecordCursorFactory {
+
+        private final Path path = new Path();
+        private final AttributeClassCatalogueCursor cursor;
+
+        public AttributeCatalogueCursorFactory(CairoConfiguration configuration, RecordMetadata metadata) {
+            super(metadata);
+            this.cursor = new AttributeClassCatalogueCursor(configuration, path);
+        }
+
+        @Override
+        public void close() {
+            Misc.free(path);
+        }
+
+        @Override
+        public RecordCursor getCursor(SqlExecutionContext executionContext) {
+            cursor.toTop();
+            return cursor;
+        }
+
+        @Override
+        public boolean recordCursorSupportsRandomAccess() {
+            return false;
+        }
+    }
+
+    private static class AttributeClassCatalogueCursor implements NoRandomAccessRecordCursor {
+        private final Path path;
+        private final FilesFacade ff;
+        private final DiskReadingRecord diskReadingRecord = new DiskReadingRecord();
+        private final NativeLPSZ nativeLPSZ = new NativeLPSZ();
+        private final int plimit;
+        private final ReadOnlyColumn metaMem;
+        private long findFileStruct = 0;
+        private int columnIndex = 0;
+        private int tableId = 1000;
+        private boolean readNextFileFromDisk = true;
+        private int columnCount;
+        private boolean hasNextFile = true;
+        private boolean foundMetadataFile = false;
+
+        public AttributeClassCatalogueCursor(CairoConfiguration configuration, Path path) {
+            this.ff = configuration.getFilesFacade();
+            this.path = path;
+            this.path.of(configuration.getRoot()).$();
+            this.plimit = this.path.length();
+            this.metaMem = new OnePageMemory();
+        }
+
+        @Override
+        public void close() {
+            if (findFileStruct != 0) {
+                ff.findClose(findFileStruct);
+                findFileStruct = 0;
+            }
+            metaMem.close();
```

Comment:
perhaps this class can also (like Path) be at factory level so we do not need to allocate to re-execute the cursor